frontend/src/scenes/insights/AdHocInsight.tsx

```diff
@@ -0,0 +1,33 @@
+import { FilterType, InsightLogicProps } from '~/types'
+import React, { useMemo } from 'react'
+import { BindLogic } from 'kea'
+import { insightLogic } from 'scenes/insights/insightLogic'
+import { useValues } from 'kea'
+import { InsightViz } from 'lib/components/InsightCard/InsightCard'
+
+let uniqueMemoizedIndex = 0
+export function AdHocInsight({
+    filters,
+    style,
+}: {
+    filters: Partial<FilterType>
+    style: React.CSSProperties
+}): JSX.Element {
+    const pageKey = useMemo(() => `filter-${uniqueMemoizedIndex++}`, [])
```

Comment:
Kinda dodgy to rely on this as `memo` isn't guaranteed to hold for re-renders, and the key changing could mess up what state the logic uses.